pkgs/by-name/to/todesk/package.nix

```
@@ -0,0 +1,147 @@
+{
+  stdenv,
+  lib,
+  procps,
+  fetchurl,
+  dpkg,
+  writeShellScript,
+  buildFHSEnv,
+  nspr,
+  kmod,
+  systemdMinimal,
+  glib,
+  pulseaudio,
+  libXext,
+  libX11,
+  libXrandr,
+  glibc,
+  cairo,
+  libdrm,
+  coreutils,
+  libXi,
+  libGL,
+  bash,
+  libXcomposite,
+  libXdamage,
+  libXfixes,
+  libXtst,
+  nss,
+  libXxf86vm,
+  gtk3,
+  gdk-pixbuf,
+  pango,
+  libz,
+  libayatana-appindicator,
+}:
+
+let
+  version = "4.7.2.0";
+  todesk-unwrapped = stdenv.mkDerivation (finalAttrs: {
+    pname = "todesk-unwrapped";
+    version = version;
+    src = fetchurl {
+      url = "https://newdl.todesk.com/linux/todesk-v${finalAttrs.version}-amd64.deb";
+      hash = "sha256-v7VpXXFVaKI99RpzUWfAc6eE7NHGJeFrNeUTbVuX+yg=";
+      curlOptsList = [
+        "--user-agent"
+        "Mozilla/5.0"
+      ];
+    };
+    nativeBuildInputs = [ dpkg ];
+
+    unpackPhase = ''
+      runHook preUnpack
+      dpkg -x $src ./todesk-src
+      runHook postUnpack
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p "$out/lib"
+      cp -r todesk-src/* "$out"
+      cp "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1" "$out/opt/todesk/bin/libappindicator3.so.1"
+      mv "$out/opt/todesk/bin" "$out/bin"
+      cp "$out/bin/libmfx.so.1" "$out/lib" # These libraries doesn't find in nixpkgs but in $out/bin as well. I don't know why doesn't work.
+      cp "$out/bin/libglut.so.3" "$out/lib"
+      mkdir "$out/opt/todesk/config"
+      mkdir "$out/opt/todesk/bin"
+      mkdir -p "$out/share/applications"
+      mkdir "$out/share/icons"
+      runHook postInstall
+    '';
+
+  });
+
+in
+buildFHSEnv {
+  inherit version;
+  name = "todesk";
+  targetPkgs = pkgs: [
+    todesk-unwrapped
+    pulseaudio
+    nspr
+    kmod
+    libXi
+    systemdMinimal
+    glib
+    libz
+    bash
+    coreutils
+    libX11
+    libXext
+    libXrandr
+    glibc
+    libdrm
+    libGL
+    procps
+    cairo
+    libXcomposite
+    libXdamage
+    libXfixes
+    libXtst
+    nss
+    libXxf86vm
+    gtk3
+    gdk-pixbuf
+    pango
+
+  ];
+  extraBwrapArgs = [
+    "--bind /var/lib/todesk /opt/todesk/config" # create the folder before bind to avoid permission denided.
+    "--bind ${todesk-unwrapped}/bin /opt/todesk/bin"
+    "--bind /home /home"
```

```suggestion
    "--bind /home /home"
```
why do we need to `bind` the home folder of all users?